scripts/run-ui-tests

```diff
@@ -17,125 +14,17 @@ source "$self_dir"/utils.sh
-  tail -n+$((ide_log_last_line+1)) $ide_log \
-    | sed '/INFO - pl.ProjectRootManagerComponent - project roots have changed/, /at java.desktop.java.awt.EventDispatchThread.run/ d'
-}
-
-# If we're on CI, run IDE on X Virtual Framebuffer; otherwise, open a regular window.
-# --auto-servernum is deprecated in favor of --auto-display,
-# but unfortunately xvfb-run available for Debian Buster (base of our CI image) doesn't support the latter.
-command=${CI:+xvfb-run --auto-servernum}
-gradle_run_ide_log=~/gradlew-runIdeForUiTests.log
-$command ./gradlew \
-  -PintellijVersionForRunIde=$intellij_version \
-  --no-daemon \
-  --max-workers=1 \
-  -Dorg.gradle.jvmargs=-Xmx256M \
-  runIdeForUiTests >$gradle_run_ide_log 2>&1 &
-
-# $! expands to the PID of the last process executed in the background.
-wrapper_process_pid=$!
-function is_wrapper_process_running() {
-  ps -p $wrapper_process_pid >/dev/null 2>&1
-}
-
-function finish() {
-  if is_wrapper_process_running; then
-    echo 'Sending TERM to the wrapper process...'
-    # Ignoring exit code in case the process just terminated between `is_wrapper_process_running` and `kill`.
-    kill -TERM $wrapper_process_pid || true
-  fi
-  # Let's wait until the wrapper process is terminated to avoid any risk of race conditions
-  # when another UI test is to be run right after this script completes.
-  local count=0
-  while is_wrapper_process_running; do
-    if (( count++ == 10 )); then
-      # Rather unlikely, wrapper process tends to shut down quickly on receiving SIGTERM.
-      echo 'Sending KILL to the wrapper process...'
-      kill -KILL $wrapper_process_pid || true
-    fi
-    echo 'Waiting for the wrapper process to shut down...'
-    sleep 1.0
-  done
-
-  # If the IDE process still occupies the port at this point, let's kill the IDE process itself.
-  if is_robot_server_listening; then
-    echo 'Sending TERM to IDE process...'
-    fuser --kill -TERM $robot_server_port/tcp >/dev/null 2>&1 || true
-  fi
-  count=0
-  while is_robot_server_listening; do
-    if (( count++ == 10 )); then
-      # Rather unlikely, IDE tends to shut down quickly on receiving SIGTERM.
-      echo 'Sending KILL to IDE process...'
-      fuser --kill -KILL $robot_server_port/tcp >/dev/null 2>&1 || true
-    fi
-    echo 'Waiting for IDE process to shut down...'
-    sleep 1.0
-  done
-}
-trap finish EXIT
-
-gradle_run_ide_log_last_line=0
-# Let's wait until Remote Robot plugin living within the IDE starts listening.
-info 'Logs of gradle runIdeForUiTests'
-until is_robot_server_listening; do
-  sleep 0.5
-  tail -n+$((gradle_run_ide_log_last_line+1)) $gradle_run_ide_log
-  gradle_run_ide_log_last_line=$(wc -l < $gradle_run_ide_log)
-  if ! is_wrapper_process_running; then
-    die 'Wrapper process has been terminated before Remote Robot plugin started listening; aborting'
-  fi
-done
-
-info 'Logs of gradle uiTests:test'
+export IDEPROBE_DISPLAY=xvfb
```

Comment:
I'd rather make it `xvfb` only if `CI` env var is `true`, otherwise just the default display